pkg/config/xds/lds.go

```diff
@@ -102,42 +103,68 @@ func (l *LdsManager) removeListeners(toRemoveHash map[string]struct{}) {
 	names := make([]string, 0, len(toRemoveHash))
 	for name := range toRemoveHash {
 		names = append(names, name)
-		l.listenerMg.RemoveListener(names)
 	}
+	l.listenerMg.RemoveListener(names)
 }
 
 // setupListeners setup listeners accord to dynamic resource
 func (l *LdsManager) setupListeners(listeners []*xdsModel.Listener) {
+	//Make sure each one has a unique name like "host-port-protocol"
+	for _, v := range listeners {
+		v.Name = resolveListenerName(v.Address.SocketAddress.Address, int(v.Address.SocketAddress.Port), v.Protocol.String())
```

Comment:
is it possible that v.Name  has it's value. check it is nil then call resolveListenerName ?

Reply:
The listener’s name is useless now, so just overwritten it. if we need this, then change it later, but not now.